compiler/noirc_evaluator/src/ssa/opt/mem2reg/block.rs

```diff
@@ -93,16 +73,25 @@ impl Block {
             // Now we have to invalidate every reference we know of
             self.invalidate_all_references();
         } else if let Some(alias) = aliases.single_alias() {
-            self.references.insert(alias, value);
+            self.set_reference_value(alias, value);
```

Comment:
Nitpick: is this needed when we set the value below on line 86?

We can consider removing this case and changing the else to `else if aliases.len() > 1`

Reply:
I wanted to do exactly what you mention here. I thought "if the address has a single alias then it must be itself". I added `assert_eq!(address, alias)` and... it fails! Which means that an address has a single alias that's not itself... but I didn't investigate why that happens. I guess I'll do that now as I think that shouldn't happen, right?

Reply:
Makes sense. I guess we can merge this PR and revisit this "address has one alias but it's not itself" thing in a follow-up issue or PR.